server/test/api/accounts.js

```diff
@@ -121,6 +122,28 @@ describe('The account API, PATH : /accounts', function () {
       .catch(helpers.handler);
   });
 
+ it('METHOD : GET, PATH : /accounts/:id/balance, It returns an error 404 not found', function () {
+   return agent.get('/accounts/:id/balance'.replace(':id', FETCHABLE_ACCOUNT_ID))
+    .then(function (res){
+      expect(res).to.have.status(404);
+      expect(res).to.be.json;
+    })
+    .catch(helpers.handler);
+ });
+
+  
+  it('METHOD : GET, PATH : /accounts/:id/balance?journal=1, It returns the balance of a provided account_id, scans the journal also', function () {
+    return agent.get('/accounts/:id/balance?journal=1'.replace(':id', ACCOUNT_ID_FOR_BALANCE))
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(res.body).to.not.be.empty;
+        expect(res.body).to.have.all.keys('account_id', 'debit', 'credit', 'balance');
+        expect(res.body.debit).to.satisfy(function (debit) { return debit >= 0;});
```

Comment:
I don't think this test is necessary if the columns are `unsigned`.  MySQL will ensure they should always be positive.